extensions/smallrye-reactive-messaging-pulsar/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/pulsar/deployment/PulsarDevServicesProcessor.java

```diff
@@ -175,10 +183,21 @@ private RunningDevService startPulsarContainer(DockerStatusBuildItem dockerStatu
                 container.withPort(config.fixedExposedPort);
             }
             timeout.ifPresent(container::withStartupTimeout);
+            String hostName = null;
+            if (useSharedNetwork) {
+                hostName = ConfigureUtil.configureSharedNetwork(container, DEV_SERVICE_PULSAR);
+            }
             container.start();
 
-            return getRunningService(container.getContainerId(), container::close, container.getPulsarBrokerUrl(),
-                    container.getHttpServiceUrl());
+            var pulsarBrokerUrl = container.getPulsarBrokerUrl();
```

Comment:
I'd do all this inside the `PulsarContainer`.

Reply:
Sure, moved.